src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/BootstrapUsersServiceTests.cs

```diff
@@ -0,0 +1,171 @@
+#nullable enable
+using GovUk.Education.ExploreEducationStatistics.Admin.Models;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Fixtures;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Tests.Fixtures;
+using Microsoft.Extensions.Configuration;
+using Moq;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Tests.Services;
+
+public class BootstrapUsersServiceTests
+{
+    private readonly DataFixture _dataFixture = new();
+
+    [Theory]
+    [InlineData(null)]
+    [InlineData("")]
+    [InlineData(" ")]
+    [InlineData(",")]
+    [InlineData(" , ")]
+    [InlineData(",,")]
+    [InlineData(", ,")]
+    public void AddBootstrapUsers_EmptyEmailList_DoesNothing(string? bootstrappedEmailsString)
+    {
+        var bootstrappedUserEmailsConfiguration = new Mock<IConfiguration>(MockBehavior.Strict);
+        var bootstrapUsersSection = new Mock<IConfigurationSection>();
+
+        var bauSection = new Mock<IConfigurationSection>();
+        bauSection.Setup(s => s.Value).Returns(bootstrappedEmailsString);
+
+        bootstrapUsersSection.Setup(s => s.GetSection("BAU")).Returns(bauSection.Object);
+
+        bootstrappedUserEmailsConfiguration
+            .Setup(c => c.GetSection("BootstrapUsers"))
+            .Returns(bootstrapUsersSection.Object);
+
+        var contentDbContextId = Guid.NewGuid().ToString();
+
+        using (var contentDbContext = DbUtils.InMemoryApplicationDbContext(contentDbContextId))
+        {
+            var service = SetupService(contentDbContext, bootstrappedUserEmailsConfiguration.Object);
+
+            service.AddBootstrapUsers();
+        }
+
+        using (var contentDbContext = DbUtils.InMemoryApplicationDbContext(contentDbContextId))
+        {
+            var users = contentDbContext.Users.ToList();
+
+            Assert.Empty(users);
+        }
+    }
+
+    [Fact]
+    public void AddBootstrapUsers_UsersAlreadyExist_DoesNothing()
+    {
+        var placeholderDeletedUser = _dataFixture.DefaultUser().WithEmail(User.DeletedUserPlaceholderEmail).Generate();
+        var existingUsers = _dataFixture
+            .DefaultUser()
+            .ForIndex(0, s => s.SetEmail("test1@test.com"))
+            .ForIndex(1, s => s.SetEmail("test2@test.com"))
+            .GenerateList(2);
```

Comment:
You could use

```csharp
var (user1, user2) = _dataFixture
    .DefaultUser()
    .ForIndex(0, s => s.SetEmail("test1@test.com"))
    .ForIndex(1, s => s.SetEmail("test2@test.com"))
    .GenerateTuple2();
```

to avoid referencing them from array indexes later on, but this is personal preference.